backend/core/test/listings/listings.e2e-spec.ts

```diff
@@ -453,6 +453,158 @@ describe("Listings", () => {
     })
   })
 
+  // REMOVE_WHEN_EXTERNAL_NOT_NEEDED
+  describe("/listings/combined", () => {
+    it("should return all listings", async () => {
+      const res = await supertest(app.getHttpServer()).get("/listings/combined").expect(200)
+      expect(res.body.items.map((listing) => listing.id).length).toBeGreaterThan(0)
+    })
+
+    it("should return the first page of paginated listings", async () => {
+      // Make the limit 1 less than the full number of listings, so that the first page contains all
+      // but the last listing.
+      const page = "1"
+      // This is the number of listings in ../../src/seed.ts minus 1
+      const limit = 15
+      const params = "/?page=" + page + "&limit=" + limit.toString()
+      const res = await supertest(app.getHttpServer())
+        .get("/listings/combined" + params)
+        .expect(200)
+      expect(res.body.items.length).toEqual(limit)
+    })
+
+    it("should return the last page of paginated listings", async () => {
+      // Make the limit 1 less than the full number of listings, so that the second page contains
+      // only one listing.
+      // query to get max number of listings
+      let queryParams = {
+        limit: 1,
+        page: 1,
+        view: "base",
+      }
+      let query = qs.stringify(queryParams)
+      let res = await supertest(app.getHttpServer()).get(`/listings/combined?${query}`).expect(200)
+      const totalItems = res.body.meta.totalItems
```

Comment:
Is there a reason why in this test you make a call to get the total number of listings whereas in the call above you hardcode it to 15 (16-1)? Can it be done the same way for both tests?

Reply:
That's a good question. The short answer is that I just copied the tests from the `/listings` endpoint since I figured we'd want to see the same behavior out of both to start.  You can see the original test on line 75.

My interpretation is that the first call in the test defined on line 476 is only to determine the total number of items (line 487), which it then uses on line 490 to set up another call that is designed to always get the last page of results.  The "real" test happens with the `expect` on line 496.